Core/Resource/SymbolJson.cs

```diff
@@ -368,9 +336,144 @@ public Symbol ReadSymbol(Model model, JToken o, bool allowNonOperatorInstanceTyp
+            {
+                Log.Error($"Skipping child of undefined type {childIdString} in {name}");
+                return;
+            }
+            
+            missingSymbolChildIds.Add(childId);
+        }
+        
+        // returns an id with valid C# namespace characters ('_' instead of '-') and all-lowercase
+        static string CreateGuidNamespaceString(Guid guid)
+        {
+            var idSpan = guid.ToString().AsSpan();
+            Span<char> namespaceIdSpan = stackalloc char[idSpan.Length];
+            idSpan.ToLowerInvariant(namespaceIdSpan);
+
+            int index;
+            while ((index = namespaceIdSpan.IndexOf('-')) != -1)
+            {
+                namespaceIdSpan[index] = '_';
+            }
+
+            return new string(namespaceIdSpan);
+        }
+
+        // Method for when allowNonOpInstanceType = true
+        static Type GetOperatorInstanceType(string typeName)
+        {
+            bool typeIsCached = TypeCache.TryGetValue(typeName, out var type);
+
+            if (!typeIsCached)
+            {
+                type = Type.GetType(typeName);
+                TypeCache.TryAdd(typeName, type);
+            }
+
+            if (type is not null)
+                return type;
+                
+            MessageBox.Show($"Definition '{typeName}' is missing in Operator.dll.\nPlease try to rebuild your solution.");
+            Application.Exit();
+            Application.ExitThread();
+
+            return null;
+        }
+        
+        // Method for when allowNonOpInstanceType = false
+        static Type GetAnyInstanceType(string typeName)
+        {
+            var typeIsCached = TypeCache.TryGetValue(typeName, out var type);
+            if (typeIsCached)
+            {
+                return type ?? ObjectType;
+            }
+
+            type = Type.GetType(typeName) ?? ObjectType;
+            TypeCache.TryAdd(typeName, type);
+
+            return type;
         }
+        
+        private static void ReadConnections(JArray connectionsJson, string name, HashSet<Guid> missingSymbolChildIds, List<Symbol.Connection> connections)
+        {
+            foreach (var c in connectionsJson)
+            {
+                var connection = ReadConnection(c);
+
+                var undefinedType = missingSymbolChildIds.Contains(connection.TargetParentOrChildId)
+                                    || missingSymbolChildIds.Contains(connection.SourceParentOrChildId);
+
+                if (undefinedType)
+                {
+                    Log.Warning($"Skipping connection to child of undefined type in {name}");
+                    continue;
+                }
+
+                connections.Add(connection);
+            }
+        }
+
+        private static readonly Type ObjectType = typeof(object);
+        // Define a static dictionary to cache Type instances by instanceTypeName
+        private static readonly ConcurrentDictionary<string, Type> TypeCache = new();

```

Comment:
cached types to remove redundant `typeof` invokations

Reply:
feedback definitely heard, but I imagine the compilers make decisions about compile-time caching of values. I saw a noticeable performance boost from caching some types, I don't remember which, when this was a single-threaded operation. That being said, it's possible that this type caching is no longer beneficial and it should be benchmarked.

Reply:
in the new parallel workflow I'm no longer seeing as large of a performance boost for all the type caching, probably about ~100ms, so I am removing the type cache